### PR TITLE
[Improve][Connector-V2][Clickhouse] Unified exception for Clickhouse source & sink connector

### LINKED ISSUE #3549: [Improve][Connector-V2][Clickhouse] Unified exception for Clickhouse source & sink connector
sub-task of #2930

### PR DESCRIPTION
**Pull request by FWLamb**
## Purpose of this pull request

sub-task https://github.com/apache/incubator-seatunnel/issues/2930

close https://github.com/apache/incubator-seatunnel/issues/3549

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)

### CODE REVIEW

**Comment by FWLamb** [comment]
@Hisoka-X @TyrantLucifer PTAL

File: seatunnel-connectors-v2/connector-clickhouse/src/main/java/org/apache/seatunnel/connectors/seatunnel/clickhouse/exception/ClickhouseConnectorErrorCode.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.connectors.seatunnel.clickhouse.exception;
+
+import org.apache.seatunnel.common.exception.SeaTunnelErrorCode;
+
+public enum ClickhouseConnectorErrorCode implements SeaTunnelErrorCode {
+
+    FIELD_NOT_IN_TABLE("CLICKHOUSE-01", "Field is not existed in target table"),
```
Comment (EricJoy2048) [inline seatunnel-connectors-v2/connector-clickhouse/src/main/java/org/apache/seatunnel/connectors/seatunnel/clickhouse/exception/ClickhouseConnectorErrorCode.java:24]: Please add connector error code in [docs/en/connector-v2/Error-Quick-Reference-Manual.md](https://github.com/apache/incubator-seatunnel/pull/3564/files#diff-91ca769bb07009f1bc8bf5a914dc30811794261e4da18b84392ea45930701489)

File: seatunnel-connectors-v2/connector-clickhouse/src/main/java/org/apache/seatunnel/connectors/seatunnel/clickhouse/sink/client/ClickhouseSinkWriter.java [hunk 2]
```diff
@@ -122,7 +124,7 @@ public void close() throws IOException {
                     intHolder.setValue(0);
                 }
             } catch (SQLException e) {
-                throw new RuntimeException("Failed to close prepared statement.", e);
+                throw new ClickhouseConnectorException(CommonErrorCode.WRITER_OPERATION_FAILED, "Failed to close prepared statement.", e);
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-clickhouse/src/main/java/org/apache/seatunnel/connectors/seatunnel/clickhouse/sink/client/ClickhouseSinkWriter.java]: CommonErrorCode.SQL_OPERATION_FAILED is better

**Review by TyrantLucifer — COMMENTED** [review]
cc @Hisoka-X PTAL.

File: seatunnel-connectors-v2/connector-clickhouse/src/main/java/org/apache/seatunnel/connectors/seatunnel/clickhouse/sink/client/ClickhouseSinkWriter.java (hunk 2 above)
Comment (FWLamb) [reply seatunnel-connectors-v2/connector-clickhouse/src/main/java/org/apache/seatunnel/connectors/seatunnel/clickhouse/sink/client/ClickhouseSinkWriter.java]: done

File: seatunnel-connectors-v2/connector-clickhouse/src/main/java/org/apache/seatunnel/connectors/seatunnel/clickhouse/exception/ClickhouseConnectorErrorCode.java (hunk 1 above)
Comment (FWLamb) [reply seatunnel-connectors-v2/connector-clickhouse/src/main/java/org/apache/seatunnel/connectors/seatunnel/clickhouse/exception/ClickhouseConnectorErrorCode.java:24]: done

**Review by EricJoy2048 — COMMENTED** [review]
LGTM, Please resolve conflicts.

File: docs/en/connector-v2/Error-Quick-Reference-Manual.md [hunk 3]
```diff
@@ -46,7 +46,7 @@ problems encountered by users.
 | CASSANDRA-02 | Add batch SeaTunnelRow data into a batch failed | When users encounter this error code, it means that cassandra has some problems, please check it whether is work                                                      |
 | CASSANDRA-03 | Close cql session of cassandra failed           | When users encounter this error code, it means that cassandra has some problems, please check it whether is work                                                      |
 | CASSANDRA-04 | No data in source table                         | When users encounter this error code, it means that source cassandra table has no data, please check it                                                               |
-| CASSANDRA-05 | Parse ip address from string field failed       | When users encounter this error code, it means that upstream data does not match ip address format, please check it                                                   |
+| CASSANDRA-05 | Parse ip address from string field field        | When users encounter this error code, it means that upstream data does not match ip address format, please check it                                                   |
```
Comment (Hisoka-X) [inline docs/en/connector-v2/Error-Quick-Reference-Manual.md]:
```suggestion
| CASSANDRA-05 | Parse ip address from string field        | When users encounter this error code, it means that upstream data does not match ip address format, please check it                                                   |
```

**Comment by FWLamb** [comment]
> LGTM, Please resolve conflicts.

done.

File: docs/en/connector-v2/Error-Quick-Reference-Manual.md (hunk 3 above)
Comment (FWLamb) [reply docs/en/connector-v2/Error-Quick-Reference-Manual.md]: done.

**Review by Hisoka-X — APPROVED** [review]

**Review by TyrantLucifer — APPROVED** [review]

**Pull request merged**